src/FilterBuilder.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+namespace Webchain\ScimFilterToDqb;
+
+use Doctrine\ORM\Query\Expr\Composite;
+use Doctrine\ORM\QueryBuilder;
+use Tmilos\ScimFilterParser\Ast\AttributePath;
+use Tmilos\ScimFilterParser\Ast\ComparisonExpression;
+use Tmilos\ScimFilterParser\Ast\Conjunction;
+use Tmilos\ScimFilterParser\Ast\Disjunction;
+use Tmilos\ScimFilterParser\Ast\Negation;
+use Tmilos\ScimFilterParser\Ast\Node;
+use Tmilos\ScimFilterParser\Ast\ValuePath;
+use Webchain\ScimFilterToDqb\ValueObject\AttributeOperator;
+
+class FilterBuilder
+{
+    /** @var QueryBuilder */
+    private $queryBuilder;
+    /** @var Joiner  */
+    private $joiner;
+
+    private $parameters = [];
+
+    public function __construct(Joiner $joiner)
+    {
+        $this->joiner = $joiner;
+    }
+
+    public function addFiltersOnQueryBuilder(QueryBuilder $queryBuilder, Node $node)
```

Comment:
Why do you pass `QueryBuilder` in construct, because is used in both public methods

Reply:
Well, because the name of the method suggests that you add filters on a QueryBuilder so i'm expecting, from the name of the method, to have 2 parameters, the QueryBuilder and the filters, which they are :P